tests/test-docker-image.sh

```
@@ -53,11 +53,26 @@ echo "$VERSIONS" | grep -q "Python 3" && pass "Python 3" || fail "Python version
+# module importable and changes the API or the output shape, so a bare require()
+# waves it through; parsing a heading and checking for <h1> does not.
+RESULT=$(run_in_container "node -e \"const m=require('marked'); const p=(m.parse||m.marked); const h=p('# t'); console.log(/<h1/.test(h)?'OK':'FAIL')\"") || RESULT=""
+echo "$RESULT" | grep -q "OK" && pass "marked renders a heading" || fail "marked render"
+
+# tsc COMPILES. The CLI loop below only proves --version exits 0, which a major
+# that changed type-checking would also satisfy. This compiles a strict file and
+# then asserts a deliberately ill-typed one is REJECTED, so the check cannot pass
+# by accepting everything.
+RESULT=$(run_in_container "d=\$(mktemp -d); printf 'export const twice = (x: number): number => x * 2;\\n' > \$d/a.ts; tsc --noEmit --strict \$d/a.ts >/dev/null 2>&1 && echo OK || echo FAIL") || RESULT=""
+echo "$RESULT" | grep -q "OK" && pass "tsc compiles a strict file" || fail "tsc compile"
+RESULT=$(run_in_container "d=\$(mktemp -d); printf 'const n: number = \\"nope\\";\\n' > \$d/b.ts; tsc --noEmit --strict \$d/b.ts >/dev/null 2>&1 && echo FAIL || echo OK") || RESULT=""
```

Shellcheck (0.11.0)

[warning] Word is of the form "A"B"C" (B indicated). Did you mean "ABC" or "A\"B\"C"?

(SC2140)